### PR TITLE
feat: add webhooks for workos sync integration

### PR DESCRIPTION
**Pull request by tgmendes**
Two things are achieved by this PR:

1. We wire in the webhooks endpoint so when the time comes we can start sending them (we won't have signing secret yet so nothing will happen until that + workOS is configured)
2. Update the code to properly handle org memberships (previously memberships were treated as user events and updated the users table - this was wrong)
3. Also adds handling of global roles which was missing

## Important

This is not yet fully ready to turn on. Operationally it works fine for most normal use cases, but there are some edge cases around eventual consistency we need to get right first.

I've added a summary of the different failure modes and recovery scenarios as well as a thorough test plan to verify that this is working properly.

## Follow-up work

* Start tracking user updates
* Better handling of membership / role assignments when data doesn't exist yet (this will require some DB migrations)
* Implement failure mode recovery
* Backfill and reconcilliation scripts

---

<a href="https://app.devin.ai/review/speakeasy-api/gram/pull/2690" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Review by adaam2 — APPROVED** [review]

File: server/internal/background/activities/process_workos_global_role_events.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	"github.com/speakeasy-api/gram/server/internal/thirdparty/workos"
+	workosrepo "github.com/speakeasy-api/gram/server/internal/thirdparty/workos/repo"
+)
+
+const workosGlobalRoleEventsPageSize = 100
+
+// WorkosGlobalRoleCursorKey is the workos_organization_id value used to store
+// the singleton cursor for environment-level role events in workos_organization_syncs.
+// Empty string is intentional: role.* events have no organization scope.
+const WorkosGlobalRoleCursorKey = ""
```
[the remark this answers is not in the document]
Comment (tgmendes) [reply server/internal/background/activities/process_workos_global_role_events.go:29]: This is OK and the table accounts for this.
